crypto/src/proof_of_work.rs

```diff
@@ -0,0 +1,152 @@
+use num_bigint::BigUint;
+use hex::{FromHex, FromHexError};
+use std::convert::TryFrom;
+use sodiumoxide::randombytes::randombytes;
+use super::{blake2b, crypto_box::PublicKey};
+
+#[derive(Clone, PartialEq)]
+pub struct ProofOfWork([u8; 0x18]);
```

Comment:
@vlad9486 
PR looks cool :),

there are several usage of 0x18 / 0x20, what if you introduce those as constants?
I dont know, if it will be readable than?

at least, some short comments/description should here, where you mention 0x18 / 0x20, that they are used as indexes or "number of bytes" or something like that